### PR TITLE
zigbee: Move scenes extension to a separate library

### PR DESCRIPTION
**Pull request by maciekfabia**
Create a separate zigbee library from scenes extension
that was available as part of light bulb sample by now.

### CODE REVIEW

**Review by rlubos — APPROVED** [review]
LGTM

**Comment by maciekfabia** [comment]
I started additional CI related to scenes functionality. I'll remove DNM after it succeeds.

File: subsys/zigbee/lib/zigbee_scenes/Kconfig [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#
+# Copyright (c) 2021 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+menuconfig ZIGBEE_SCENES
+	bool "Enable Zigbee scenes extension"
+	depends on SETTINGS
```
Comment (sebastiandraus) [inline subsys/zigbee/lib/zigbee_scenes/Kconfig:9]:
```suggestion
	select SETTINGS
	imply NVS
```

File: samples/zigbee/light_bulb/prj_nrf5340dk_nrf5340_cpuapp.conf [hunk 2]
```diff
@@ -42,6 +42,7 @@ CONFIG_NET_IPV6_RA_RDNSS=n
 CONFIG_NET_IP_ADDR_CHECK=n
 CONFIG_NET_UDP=n
 
-#settings storage
+# Scene extension requires settings
 CONFIG_NVS=y
 CONFIG_SETTINGS=y
```
Comment (sebastiandraus) [inline samples/zigbee/light_bulb/prj_nrf5340dk_nrf5340_cpuapp.conf:47]: Leave selecting SETTINGS and SETTINGS backend to the library
```suggestion
```

File: samples/zigbee/light_bulb/prj.conf [hunk 3]
```diff
@@ -43,6 +43,7 @@ CONFIG_NET_IPV6_RA_RDNSS=n
 CONFIG_NET_IP_ADDR_CHECK=n
 CONFIG_NET_UDP=n
 
-#settings storage
+# Scene extension requires settings
 CONFIG_NVS=y
 CONFIG_SETTINGS=y
```
Comment (sebastiandraus) [inline samples/zigbee/light_bulb/prj.conf:48]: Leave selecting SETTINGS and SETTINGS backend to the library
```suggestion
```

File: samples/zigbee/light_bulb/prj.conf [hunk 4]
```diff
@@ -43,6 +43,7 @@ CONFIG_NET_IPV6_RA_RDNSS=n
 CONFIG_NET_IP_ADDR_CHECK=n
 CONFIG_NET_UDP=n
 
-#settings storage
+# Scene extension requires settings
```
Comment (sebastiandraus) [inline samples/zigbee/light_bulb/prj.conf:46]: Update comment

File: samples/zigbee/light_bulb/prj_nrf5340dk_nrf5340_cpuapp.conf [hunk 5]
```diff
@@ -42,6 +42,7 @@ CONFIG_NET_IPV6_RA_RDNSS=n
 CONFIG_NET_IP_ADDR_CHECK=n
 CONFIG_NET_UDP=n
 
-#settings storage
+# Scene extension requires settings
```
Comment (sebastiandraus) [inline samples/zigbee/light_bulb/prj_nrf5340dk_nrf5340_cpuapp.conf:45]: Update comment

**Review by sebastiandraus — COMMENTED** [review]
Nit:
- empty lines for two new files
- light sample main.c order of including headers

**Review by sebastiandraus — APPROVED** [review]
LGTM

**Comment by maciekfabia** [comment]
@rlubos CI testing scenes functionality passed, I'm removing DNM label.
Ready to merge when CI gets green

**Pull request merged**